### PR TITLE
Create common api to trigger webapps backups

### LINKED ISSUE #24465: Create common API to trigger webapps Backups
Create common API to take backups of Optimize, Identity, Tasklist and Operate together. The implementation should be similar as what [Operate/Tasklist are current doing](https://github.com/camunda/camunda/blob/4cf854cb77ad186d2f9fd9ef2d2a8cc77996052c/operate/webapp/src/main/java/io/camunda/operate/webapp/management/BackupController.java#L38), but should include indices from all the components

### Acceptance Criteria
- [ ] POST /webapps should trigger a backup of all webapps indices together
- [ ] POST /webapps should return a list of all scheduled snapshots:
```
{
  "scheduledSnapshots": [
    "camunda_webapps_8.7.0_part_1_of_6",
    "camunda_webapps_8.7.0_part_2_of_6",
    "camunda_webapps_8.7.0_part_3_of_6",
    "camunda_webapps_8.7.0_part_4_of_6",
    "camunda_webapps_8.7.0_part_5_of_6",
    "camunda_webapps_8.7.0_part_6_of_6",
    "camunda_webapps_8.7.0_part_6_of_7"
  ]
}
```
- [ ] Below error codes should be returned (but not restricted only to this list)

Code | Description
-- | --
200 OK | Backup was successfully started, snapshots will be created asynchronously. List of snapshots is returned in the response body (see example above). This list must be persisted together with the backup id to be able to restore it later.
400 Bad Request | In case something is wrong with backupId, e.g. the same backup id already exists.
500 Server Error | All other errors, e.g. ES returned error response when attempting to create a snapshot.
502 Bad Gateway | Elasticsearch or OpenSearch is not accessible, the request can be retried when it is back.

### PR DESCRIPTION
**Pull request by entangled90**
## Description
- Moved operate BackupController in `dist/` module:
  - Refactored to use `WebEndpoint` similar to Zeebe's 
  - Refactored error handling logic in controller similar to Zeebe's `BackupController`
  - Added backward compatible `BackupControllerStandalone`
- All BackupService implementation only throw exception defined in webapps-module: `BackupServiceWrapper` is not needed anymore
- removed backup controller in tasklist so that the one in dist can be used 
- Removed BackupServiceElasticSearchTest as it contained the same tests of BackupControllerIT
- Added `snapshotTimeout` & `incompleteCheckTimeoutInSeconds` to `BackupProperties` in Tasklist with the same defaults as operate

As a follow-up, I will move BackupControllerIT into the `dist` module as it contains common tests for both operate & tasklist.

> Must be merged after #25212 
## Related issues

closes #24465

### CODE REVIEW

File: tasklist/qa/integration-tests/src/test/java/io/camunda/tasklist/webapp/BackupServiceElasticSearchTest.java [hunk 1]
```diff
@@ -1,821 +1,834 @@
-/*
- * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
- * one or more contributor license agreements. See the NOTICE file distributed
```
Comment (entangled90) [inline tasklist/qa/integration-tests/src/test/java/io/camunda/tasklist/webapp/BackupServiceElasticSearchTest.java:3]: This is temporarily commented as I'm checking if all the tests  cases have been ported already

**Comment by entangled90** [comment]
Blocked by a couple of fixes done in #25160

File: dist/src/main/java/io/camunda/webapps/controllers/BackupController.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.boot.actuate.endpoint.annotation.Selector;
+import org.springframework.boot.actuate.endpoint.annotation.WriteOperation;
+import org.springframework.boot.actuate.endpoint.web.WebEndpointResponse;
+import org.springframework.boot.actuate.endpoint.web.annotation.WebEndpoint;
+import org.springframework.stereotype.Component;
+
+@Component
+@WebEndpoint(id = "backup-history")
+@ProfileOperateTasklistStandalone
```
Comment (entangled90) [inline dist/src/main/java/io/camunda/webapps/controllers/BackupController.java:39]: This must still be standalone until we have completely unified the implementations

**Comment by npepinpe** [comment]
Can we document in the original issue the decision around configuration/backwards compatibility?

**Comment by entangled90** [comment]
For that I opened  issue #25410

File: webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java [hunk 3]
```diff
@@ -111,7 +112,7 @@ public void validateRepositoryExists(final String repositoryName) {
       if (isRepositoryMissingException(e)) {
         final String reason =
             String.format("No repository with name [%s] could be found.", repositoryName);
-        throw new GenericBackupException(reason);
+        throw new MissingRepositoryException(reason);
```
Comment (npepinpe) [inline webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java:115]: :+1:

File: webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java [hunk 4]
```diff
@@ -218,7 +219,7 @@ public List<GetBackupStateResponseDto> getBackups(final String repositoryName) {
       if (isRepositoryMissingException(e)) {
         final String reason =
             String.format("No repository with name [%s] could be found.", repositoryName);
-        throw new BackupRepositoryConnectionException(reason);
+        throw new MissingRepositoryException(reason);
```
Comment (npepinpe) [inline webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java:222]: :+1:

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/es/backup/es/BackupManagerElasticSearch.java [hunk 5]
```diff
@@ -433,22 +430,23 @@ private List<SnapshotInfo> findSnapshots(final Long backupId) {
...
+        throw new ResourceNotFoundException(
+            String.format("No backup with id [%s] found.", backupId), e);
       }
       if (isRepositoryMissingException(e)) {
         final String reason =
             String.format(
                 "No repository with name [%s] could be found.",
                 tasklistProperties.getBackup().getRepositoryName());
-        throw new TasklistRuntimeException(reason);
+        throw new GenericBackupException(reason);
```
Comment (npepinpe) [inline tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/es/backup/es/BackupManagerElasticSearch.java:445]: :question: Why isn't this `MissingRepositoryException`?

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/es/backup/es/BackupManagerElasticSearch.java [hunk 6]
```diff
@@ -433,22 +430,23 @@ private List<SnapshotInfo> findSnapshots(final Long backupId) {
...
       }
       final String reason =
           String.format("Exception occurred when searching for backup with ID [%s].", backupId);
-      throw new TasklistRuntimeException(reason, e);
+      throw new GenericBackupException(reason, e);
```
Comment (npepinpe) [inline tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/es/backup/es/BackupManagerElasticSearch.java:449]: :question: Why isn't this `ResourceNotFoundException`?

File: operate/qa/integration-tests/src/test/java/io/camunda/operate/webapp/BackupControllerIT.java [hunk 7]
```diff
@@ -68,20 +65,21 @@
 import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.junit4.SpringRunner;
 
+/* This class should be moved to the dist/ module as it does not really test anything operate specific
```
Comment (npepinpe) [inline operate/qa/integration-tests/src/test/java/io/camunda/operate/webapp/BackupControllerIT.java:68]: :question: This is actually valid format for Javadocs? :smile:

**Review by npepinpe — COMMENTED** [review]
:rocket: 

- :question: Generally, do we need `GenericBackupException`? Why not use the super type `BackupException` for generic purposes?

Anyway, can pretty much approve, but I'm just not sure about both controllers requiring standalone :thinking: Why?

File: operate/qa/integration-tests/src/test/java/io/camunda/operate/webapp/BackupControllerIT.java (hunk 7 above)
Comment (entangled90) [reply operate/qa/integration-tests/src/test/java/io/camunda/operate/webapp/BackupControllerIT.java:68]: Well it was not meant to be a Javadocs, just a temporary comment

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/es/backup/es/BackupManagerElasticSearch.java (hunk 6 above)
Comment (entangled90) [reply tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/es/backup/es/BackupManagerElasticSearch.java:449]: :shrug:  I just mapped to its counterparty. This class will be deleted very soon

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/es/backup/es/BackupManagerElasticSearch.java (hunk 5 above)
Comment (entangled90) [reply tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/es/backup/es/BackupManagerElasticSearch.java:445]: :shrug:  I just mapped to its counterparty. This class will be deleted very soon

File: dist/src/main/java/io/camunda/webapps/controllers/BackupController.java (hunk 2 above)
Comment (entangled90) [reply dist/src/main/java/io/camunda/webapps/controllers/BackupController.java:39]: Just spring startup issues, I'll be removing the standalone very soon when I'll be removing all the implementations of BackupServices from tasklist and move all the startup code to `dist`

Comment (npepinpe) [reply dist/src/main/java/io/camunda/webapps/controllers/BackupController.java:39]: Alright, no need to block for this, but I'm still curious to understand it :smile:

**Review by npepinpe — APPROVED** [review]

**Pull request merged**